### PR TITLE
feat: look for firmware in $HOME/.local/share/qemu

### PR DESCRIPTION
**Pull request by jcdickinson**
### Background

If nix-darwin (no FHS) is being used to install qemu, but not lima, then figuring out what's going wrong with the firmware can get tricky.

### Changes

This allows search in `$HOME/.local/share/qemu` for the firmware. It also hints to the user that they could try copying the firmware there.

Signed-off-by: Jonathan Dickinson <jdickinson@cmgx.io>

### CODE REVIEW

File: pkg/qemu/qemu.go [hunk 1]
```diff
@@ -587,5 +596,5 @@ func getFirmware(qemuExe string, arch limayaml.Arch) (string, error) {
 	if arch == limayaml.X8664 {
 		return "", fmt.Errorf("could not find firmware for %q (hint: try setting `firmware.legacyBIOS` to `true`)", qemuExe)
 	}
-	return "", fmt.Errorf("could not find firmware for %q", qemuExe)
+	return "", fmt.Errorf("could not find firmware for %q (hint: try copying the firmware to $HOME/.local/qemu/)", qemuExe)
```
Comment (AkihiroSuda) [inline pkg/qemu/qemu.go:599]: Perhaps:
```suggestion
	return "", fmt.Errorf("could not find firmware for %q (hint: try copying the \"edk-%s-code.fd\" firmware to $HOME/.local/share/qemu/)", arch, qemuExe)
```

File: pkg/qemu/qemu.go [hunk 2]
```diff
@@ -554,11 +555,19 @@ func getAccel(arch limayaml.Arch) string {
...
+	binDir := filepath.Dir(qemuExe)                              // "/usr/local/bin"
+	localDir := filepath.Dir(binDir)                             // "/usr/local"
+	userLocalDir := filepath.Join(currentUser.HomeDir, ".local") // "$HOME/.local"
 
+	relativePath := fmt.Sprintf("share/qemu/edk2-%s-code.fd", arch)
 	candidates := []string{
-		filepath.Join(localDir, fmt.Sprintf("share/qemu/edk2-%s-code.fd", arch)), // macOS (homebrew)
+		filepath.Join(localDir, relativePath),     // macOS (homebrew)
+		filepath.Join(userLocalDir, relativePath), // XDG-like
 	}
```
Comment (AkihiroSuda) [inline pkg/qemu/qemu.go:571]: Probably `~/.local/...` should be prioritized over `/usr/local/...`

**Comment by AkihiroSuda** [comment]
Needs rebase

**Comment by jcdickinson** [comment]
Not sure how these notifications fell through, sorry. I have implemented the changes you asked for and rebased.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**